vamb/__main__.py

```diff
@@ -1195,48 +1208,86 @@ def cluster_and_write_files(
-            }
+        unsplit_path = Path(base_clusters_name + "_unsplit.tsv")
+        split_path = Path(base_clusters_name + "_split.tsv")
+        with (
+            open(unsplit_path, "a") as unsplit_clusters_file,
+            open(split_path, "a") as split_clusters_file,
+        ):
             print(
-                str(i + 1),
-                None if cluster.radius is None else round(cluster.radius, 3),
-                (
-                    None
-                    if cluster.observed_pvr is None
-                    else round(cluster.observed_pvr, 2)
-                ),
-                cluster.kind_str,
-                sum(sequence_lens[i] for i in cluster.members),
-                len(cluster.members),
-                sequence_names[cluster.medoid],
-                file=file,
-                sep="\t",
+                "name\tradius\tpeak valley ratio\tkind\tbp\tncontigs\tmedoid", file=file
             )
+            num_contigs = latent.shape[0]
+            progress_step = num_contigs / 10
+            comparer = progress_step
+            progress = 0
+            processed_contigs = 0
+            total_unsplit = 0
+            total_split = 0
+            for i, cluster in enumerate(clusters):
+                cluster_members = {
+                    sequence_names[cast(int, i)] for i in cluster.members
+                }
+                header = i == 0
+                n_unsplit_clusters, n_split_clusters = export_binning_results(
+                    fasta_output,
+                    bin_prefix,
+                    binsplitter,
+                    base_clusters_name,
+                    {i: cluster_members},
+                    sequence_names,
+                    cast(Sequence[int], sequence_lens),
+                    header,
+                    unsplit_clusters_file,
+                    split_clusters_file,
+                )
 
-    elapsed = round(time.time() - begintime, 2)
+                print(
+                    str(i + 1),
+                    None if cluster.radius is None else round(cluster.radius, 3),
+                    (
+                        None
+                        if cluster.observed_pvr is None
+                        else round(cluster.observed_pvr, 2)
+                    ),
+                    cluster.kind_str,
+                    sum(sequence_lens[i] for i in cluster.members),
+                    len(cluster_members),
+                    sequence_names[cluster.medoid],
+                    file=file,
+                    sep="\t",
+                )
+                total_unsplit += n_unsplit_clusters
+                total_split += n_split_clusters
+                processed_contigs += len(cluster_members)
+                if processed_contigs >= comparer:
+                    comparer += progress_step
+                    progress += 10
+                    logger.info(f"{progress}% of contigs clustered")
```

Comment:
The logic here is brittle. For example, what if a single bin has 25% of all contigs? Then comparer still only increments by 10%.
Instead, make a variable called "next_reporting_threshold" - a better name for "comparer". When updating it, update it to be the next 10% higher. E.g. if we moved from 15% of contigs to 32% of contigs, update it to be 40%. You can achieve this with `-((processed_contigs * 10 + 1) // - num_contigs) / 10`.
Also, fix up the logic for the `progress` variable: If a single bin brings is from e.g. 9% to 32%, it needs to print the update for 10, 20 and 30 in one go.